File: ANGEN/SampleFramework.Controls/SmoothControls - User's Guide/Samples/Snap UI Category/Fake IconEditor/VB/FakeIconEditor/ColorPanel.vb
Imports System.ComponentModel

<System.ComponentModel.DesignerCategory("Code")> _
Friend NotInheritable Class ColorPanel
    <DefaultValue(GetType(Color), "Window")> _
    Public Overrides Property BackColor() As System.Drawing.Color
        Get
            Return MyBase.BackColor
        End Get
        Set(ByVal value As System.Drawing.Color)
            MyBase.BackColor = value
        End Set
    End Property

    Private _selectedColor As Color

    Public Property SelectedColor() As Color
        Get
            If (Me._selectedColor = Color.Empty) Then
                Return Color.Transparent
            End If
            Return Me._selectedColor
        End Get
        Set(ByVal value As Color)
            If (Me._selectedColor <> value) Then
                Me._selectedColor = value
                MyBase.Invalidate()
            End If
        End Set
    End Property

    Protected Overrides Sub OnPaint(ByVal e As System.Windows.Forms.PaintEventArgs)
        MyBase.OnPaint(e)

        If Me.Width > 0 AndAlso Me.Height > 0 Then
            Using sb As New SolidBrush(Me.SelectedColor)
                e.Graphics.FillRectangle(sb, Me.ClientRectangle)
            End Using
        End If
    End Sub

    Sub New()
        Me.InitializeComponent()

        Me.SetStyle(ControlStyles.AllPaintingInWmPaint, True)
        Me.SetStyle(ControlStyles.OptimizedDoubleBuffer, True)
        Me.SetStyle(ControlStyles.UserPaint, True)

        Me.BackColor = SystemColors.Window
    End Sub
End Class

File: ANGEN/SampleFramework.Controls/SmoothControls - User's Guide/Samples/Snap UI Category/Fake IconEditor/CS/FakeIconEditor/ColorPanel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;

namespace FakeIconEditor
{
	[System.ComponentModel.DesignerCategory("Code")]
	internal sealed partial class ColorPanel : UserControl
	{
		[DefaultValue(typeof(Color), "Window")]
		public override Color BackColor
		{
			get
			{
				return base.BackColor;
			}
			set
			{
				base.BackColor = value;
			}
		}

		private Color _selectedColor;

		public Color SelectedColor
		{
			get
			{
				if (_selectedColor == Color.Empty)
				{
					return Color.Transparent;
				}
				
				return _selectedColor;
			}
			set
			{
				if (_selectedColor != value)
				{
					_selectedColor = value;
					this.Invalidate();
				}
			}
		}

		protected override void OnPaint(PaintEventArgs e)
		{
			base.OnPaint(e);

			if (this.Width > 0 && this.Height > 0)
			{
				using (SolidBrush sb = new SolidBrush(this.SelectedColor))
				{
					e.Graphics.FillRectangle(sb, this.ClientRectangle);
				}
			}
		}

		public ColorPanel()
		{
			this.InitializeComponent();
			
			this.SetStyle(ControlStyles.AllPaintingInWmPaint, true);
			this.SetStyle(ControlStyles.OptimizedDoubleBuffer, true);
			this.SetStyle(ControlStyles.UserPaint, true);

			this.BackColor = SystemColors.Window;
		}
	}
}
